[4.0] Child templates consistency

dgrammatiko opened the pull request:
Pull Request supplementary for Issue #30384 .

### Summary of Changes

This PR ensures that all the `render()` have the appropriate params (in particular `templateInherits` that acts as a switch for child templates)

### Testing Instructions

Error pages render correctly (even for child templates, the previous PR was covering only the legacy mode)

### Actual result BEFORE applying this Pull Request



### Expected result AFTER applying this Pull Request



### Documentation Changes Required


@wilsonge

dgrammatiko committed:
Handle templateInherits in all Apps correctly

richard67 commented:
API tests failing in Drone for this PR. Not sure, but it could be related to the PR. See https://ci.joomla.org/joomla/joomla-cms/34749/1/30

```
<!-- Argument 1 passed to Joomla\CMS\WebAsset\WebAssetRegistry::addTemplateRegistryFile() must be of the type string, null given, called in /tests/www/test-install/libraries/src/Error/Renderer/HtmlRenderer.php on line 55 (500 Whoops, looks like something went wrong.) -->
```

dgrammatiko committed:
API getTemplate() should retrun object

Same as the installation Application: https://github.com/joomla/joomla-cms/blob/32260311c0b6cdcc4b4686649ede1e958b9120ce/installation/src/Application/InstallationApplication.php#L401-L415

dgrammatiko commented:
@richard67 if I got this right then this was a problem in the API Application and https://github.com/joomla/joomla-cms/pull/30387/commits/15b510e71fa4c8fb88e113e36592f931fcbe7774 should fix it

SharkyKZ commented on libraries/src/Application/ApiApplication.php:
```diff
@@ -186,6 +186,17 @@ protected function respond($options = array())
+			$template->template = 'system';
+			$template->params = new Registry;
+			$template->inheritable = 0;
+			$template->parent = null;
```
Should this be null or an empty string?

https://github.com/joomla/joomla-cms/blob/0ebb8a62ea6b9569db0a452b132907a85541ac71/libraries/src/Application/CMSApplication.php#L572

dgrammatiko replied:
Good catch

dgrammatiko committed:
Update ApiApplication.php

dgrammatiko committed:
Update InstallationApplication.php

wilsonge commented:
Thanks!
